sources/cpu/layout7/cpu_constraint_poly.move

```diff
@@ -0,0 +1,2281 @@
+module verifier_addr::cpu_constraint_poly {
+
+    use std::vector;
+    use std::vector::{push_back, borrow, borrow_mut};
+
+    const EPRODUCT_INVERSE_ZERO: u64 = 0x0001;
+
+    const PRIME: u256 = 0x800000000000011000000000000000000000000000000000000000000000001;
+
+    public fun fmul(a: u256, b: u256): u256 {
+        0
+    }
+
+    public fun fexp(a: u256, b: u256): u256 {
+        0
+    }
+
+    public fun fallback(ctx: &vector<u256>): u256 {
+        let ctx = *ctx;
+        let res = 0;
+
+        let remain = 404 - vector::length(&ctx);
+
+        for (i in 0..remain) {
+            push_back(&mut ctx, 0);
+        };
+
+        {
+            // compute expmods
+            // expmods[0] = point^(trace_length / 2048)
+            {
+                let val = fexp(/*point*/ *borrow(&ctx, 34), (/*trace_length*/ *borrow(&ctx, 7) / 2048));
```

Comment:
@0x5ea000000 can we create variables for *borrow(&ctx, 34), *borrow(&ctx, 7)? These values are used multiple times